### PR TITLE
Bug 1891666: LOG-900: Updating hacking docs and csv to be more up-to-date

### PR DESCRIPTION
**Pull request by ewolinetz**
### Description
This PR:
* Updates the csv example and correcting the Provider to match CLO.
* Updates the docs/hacking.md file to be more relevant to the current development and deployment process with make targets
* Minor typo in the pull_request_template

/cc @blockloop @periklis 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-900
- Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1891666

### CODE REVIEW

File: docs/HACKING.md [hunk 1]
```diff
@@ -146,68 +60,47 @@ You do not need to set the spec.tls.key, spec.tls.certificate and spec.tls.caCer
...
 
-* the operator-sdk installed (e.g. `make operator-sdk`)
-* the operator image is built (e.g. `make image`) and available to the OKD cluster
-
-**Note:** It is necessary to set the `IMAGE_ELASTICSEARCH_OPERATOR` environment variable to a valid pull spec in order to run this test against local changes to the `elasticsearch-operator`. For example:
-```
-make deploy-image && \
-IMAGE_ELASTICSEARCH_OPERATOR=quay.io/openshift/origin-elasticsearch-operator:latest make test-e2e
-```
+* the Elasticsearch Operator is already deployed and running
```
Comment (periklis) [inline docs/HACKING.md:91]: It also assumes that the operator registry is also deployed and running

File: docs/HACKING.md [hunk 2]
```diff
@@ -146,68 +60,47 @@ You do not need to set the spec.tls.key, spec.tls.certificate and spec.tls.caCer
...
 
 ## Dev Testing
-You should first ensure that you have commands such as `imagebuilder` and `operator-sdk`
-available by using something like `https://github.com/openshift/origin-aggregated-logging/blob/master/hack/sdk_setup.sh`.
+You should first ensure that you have commands such as `podman` and `operator-sdk`
+available.
```
Comment (periklis) [inline docs/HACKING.md:95]: We don't require `operator-sdk` installed oin your host anymore, it is provided by the make targets.

File: manifests/4.6/elasticsearch-operator.v4.6.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -31,54 +31,84 @@ metadata:
...
+                    "memory": "1Gi"
+                  }
+                }
+              },
+              "nodes": [
+                {
+                  "nodeCount": 1,
+                  "roles": ["client", "data", "master"],
+                  "storage": {
+                    "size": "200G"
```
Comment (periklis) [inline manifests/4.6/elasticsearch-operator.v4.6.0.clusterserviceversion.yaml:58]: Maybe lower `200G` here? I remember @pmoogi-redhat having trouble here on IBM Cloud. IRC 80G was ok.

**Review by periklis — COMMENTED** [review]
LGTM, minor changes. Keep In mind that you may need to apply the work in CSV after #529 is merged.

**Comment by ewolinetz** [comment]
/hold
will need to go in after https://github.com/openshift/elasticsearch-operator/pull/529

**Review by blockloop — APPROVED** [review]
/lgtm

**Pull request merged**